pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -578,16 +672,119 @@ func TestFindPodVolumes(t *testing.T) {
 		if unboundSatisfied != scenario.expectedUnbound {
 			t.Errorf("Test %q failed: expected unboundSatsified %v, got %v", name, scenario.expectedUnbound, unboundSatisfied)
 		}
-		testEnv.validatePodCache(t, name, testNode.Name, scenario.pod, scenario.expectedBindings)
+		testEnv.validatePodCache(t, name, testNode.Name, scenario.pod, scenario.expectedBindings, nil)
+	}
+}
+
+func TestFindPodVolumesWithProvisioning(t *testing.T) {
+	scenarios := map[string]struct {
+		// Inputs
+		pvs     []*v1.PersistentVolume
+		podPVCs []*v1.PersistentVolumeClaim
+		// If nil, use pod PVCs
+		cachePVCs []*v1.PersistentVolumeClaim
+		// If nil, makePod with podPVCs
+		pod *v1.Pod
+
+		// Expected podBindingCache fields
+		expectedBindings   []*bindingInfo
+		expectedProvisions []*v1.PersistentVolumeClaim
+
+		// Expected return values
+		expectedUnbound bool
+		expectedBound   bool
+		shouldFail      bool
+	}{
+		"one-provisioned": {
+			podPVCs:            []*v1.PersistentVolumeClaim{provisionedPVC},
+			expectedProvisions: []*v1.PersistentVolumeClaim{provisionedPVC},
+			expectedUnbound:    true,
+			expectedBound:      true,
+		},
+		"two-unbound-pvcs,one-matched,one-provisioned": {
+			podPVCs:            []*v1.PersistentVolumeClaim{unboundPVC, provisionedPVC},
+			pvs:                []*v1.PersistentVolume{pvNode1a},
+			expectedBindings:   []*bindingInfo{binding1a},
+			expectedProvisions: []*v1.PersistentVolumeClaim{provisionedPVC},
+			expectedUnbound:    true,
+			expectedBound:      true,
+		},
+		"one-bound,one-provisioned": {
+			podPVCs:            []*v1.PersistentVolumeClaim{boundPVC, provisionedPVC},
+			pvs:                []*v1.PersistentVolume{pvBound},
+			expectedProvisions: []*v1.PersistentVolumeClaim{provisionedPVC},
+			expectedUnbound:    true,
+			expectedBound:      true,
+		},
+		"immediate-unbound-pvc": {
```

Comment:
Can you also add a test case with immediate-bound and delayed-provision

Reply:
Done